src/Lachain.Crypto/DefaultCrypto.cs

```diff
@@ -91,6 +91,8 @@ public bool VerifySignatureHashed(byte[] messageHash, byte[] signature, byte[] p
 
                 var parsedSig = new byte[65];
                 var recId = (RestoreEncodedRecIdFromSignatureBuffer(signature) - 36) / 2 / TransactionUtils.ChainId (useNewChainId);
+                if (recId < 0 || recId > 3)
```

Comment:
If the recId is 2 or 3, could it be an error? Because the way we decode and encode recId, I think it would be only 0 or 1.